homeassistant/helpers/template.py

```diff
@@ -76,7 +77,10 @@ def render_complex(value: Any, variables: TemplateVarsType = None) -> Any:
     if isinstance(value, list):
         return [render_complex(item, variables) for item in value]
     if isinstance(value, dict):
-        return {key: render_complex(item, variables) for key, item in value.items()}
+        return {
+            render_complex(key, variables): render_complex(item, variables)
```

Comment:
How does one define a template as a key? There is no validator that is converting keys to templates right now.

Reply:
I just thought about simple cases like using a variable:
```python
    data_template:
      entity_id: media_player.kef_ls50
      "{{ key_name }}": "{{ value }}"
```
Not sure how to go about validation...

Reply:
Is [this](https://github.com/home-assistant/core/pull/39008/commits/ef36e2492e30e63ded4c334e4054a76519e71dec) sufficient for the key validation?